nullaway/src/test/java/com/uber/nullaway/jspecify/JSpecifyVarargsTests.java

```diff
@@ -0,0 +1,432 @@
+            "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "public class Test {",
+            "  public void testNullableVarargsArray(Object o1, Object o2, Object o3, @Nullable Object o4) {",
+            "    Utilities.takesNullableVarargsArray(o1, o2, o3, o4);",
+            "    Utilities.takesNullableVarargsArray(o1);", // Empty var args passed
+            "    Utilities.takesNullableVarargsArray(o1, (java.lang.Object) null);",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter '(java.lang.Object[]) null' where @NonNull",
+            "    Utilities.takesNullableVarargsArray(o1, (java.lang.Object[]) null);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseAndDeclarationOnBoth() {
+    makeHelper()
+        .addSourceLines(
+            "Nullable.java",
+            "package com.uber;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Target;",
+            "@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.TYPE_USE})",
+            "public @interface Nullable {}")
+        .addSourceLines(
+            "Utilities.java",
+            "package com.uber;",
+            "public class Utilities {",
+            " public static String takesNullableVarargsArray(Object o, @Nullable Object @Nullable... others) {",
+            "  String s = o.toString() + \" \";",
+            "  // BUG: Diagnostic contains: enhanced-for expression others is @Nullable",
+            "  for (Object other : others) {",
+            "    // BUG: Diagnostic contains: dereferenced expression other is @Nullable",
+            "    s += other.toString();",
+            "  }",
+            "  return s;",
+            " }",
+            "}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "public class Test {",
+            "  public void testNullableVarargsArray(Object o1, Object o2, Object o3, @Nullable Object o4) {",
+            "    Utilities.takesNullableVarargsArray(o1, o2, o3, o4);",
+            "    Utilities.takesNullableVarargsArray(o1);", // Empty var args passed
+            "    Utilities.takesNullableVarargsArray(o1, (java.lang.Object) null);",
+            "    Utilities.takesNullableVarargsArray(o1, (java.lang.Object[]) null);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void varargsPassArrayAndOtherArgs() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.checkerframework.checker.nullness.qual.Nullable;",
+            "public class Test {",
+            "  static void takesVarargs(Object... args) {}",
+            "  static void test(Object o) {",
+            "    Object[] x = new Object[10];",
+            "    takesVarargs(x, o);",
```

Comment:
Again, why isn't this an error in JSpecify mode? Isn't the initialization here an array of 10 `null` references?

Reply:
Yes, this is again due to unsound handling of array initializers, see previous comment